src/test/java/ch/uzh/soprafs22/groupmatcher/service/MatcherServiceTest.java

```diff
@@ -186,4 +188,33 @@ void submitStudentAnswers_invalid(){
         given(answerRepository.findByIdInAndQuestion_Matcher_Id(answerIds, testMatcher.getId())).willReturn(List.of());
         assertThrows(ResponseStatusException.class,() -> matcherService.submitStudentAnswers(matcherId, studentEmail, answerIds));
     }
+
+    @Test
+    void queryAnswerMatrix_successful() {
+
+        int numStudents = 3;
+        int numAnswers = 5;
+
+        Set<Student> testStudentSet = TestingUtils.createStudents(3);
+        List<Long> testStudentIdList = testStudentSet.stream().map(Student::getId).toList();
+        Question testQuestion = TestingUtils.createQuestion(1L, 5);
+
+        given(studentRepository.getAllStudentsIdByMatcherId(anyLong()))
+                .willReturn(testStudentIdList);
+        given(answerRepository.findByQuestion_Matcher_IdAndQuestion_QuestionTypeOrderByIdAsc(anyLong(),any(QuestionType.class)))
+                .willReturn(testQuestion.getAnswers());
+        given(studentRepository.findById(anyLong()))
+                .willReturn(testStudentSet.stream().findFirst());
+        given(studentRepository.existsByIdAndSelectedAnswers_Id(anyLong(),anyLong()))
+                .willReturn(true);
+
+        double[][] studentAnswerMatrix = matcherService.createAnswerMatrixByQuestionType(1L, SINGLE_CHOICE);
+        double[][] queryAnswerMatrixAll = matcherService.createAnswerMatrixAll(1L);
+
+        assertEquals(numStudents, studentAnswerMatrix.length);
```

Comment:
You can hard-code the expected matrix and then compare the results with assertArrayEquals

Reply:
Thanks for your feedback :) I will consider it later~